### ISSUE TITLE
Github oAuth not working when email is not present

### ISSUE THREAD

**Issue by scott** [helpyio/helpy#170]
It appears that github may not always be sending an email address, which is causing the login by oauth to not work.  Also, twitter oauth never sends an email and currently Helpy uses a cheap workaround which needs to be properly handled by asking the user to provide their email address. 

Some more information on this here: http://stackoverflow.com/questions/35373995/github-user-email-is-null-despite-useremail-scope

The bottom line is Helpy needs to react properly if the email is not present in the oAuth response, and request that the user provide their email address.

First reported in #126
